beancount-tests.el

```diff
@@ -348,3 +351,95 @@ known option nmaes."
       (goto-char 0)
       (beancount-date-down-day)
       (should (equal (thing-at-point 'line) "2024-05-10\n"))))
+
+;;; Xref backend
+
+(defun beancount-test-xref-definition-pos (identifier position)
+  (let ((defs (xref-backend-definitions 'beancount identifier)))
+    (should (equal (length defs) 1))
+    (let* ((def (car (xref-backend-definitions 'beancount identifier)))
+           (loc (xref-item-location def))
+           ;; Pre-28.1 defclass was used for xref-buffer-location
+           (pos (if (version< emacs-version "28.1")
+                    (oref loc position)
+                  (xref-buffer-location-position loc))))
+      (should (equal pos position)))))
+
+(ert-deftest beancount/xref-backend-definitions ()
+  :tags '(xref)
+  (with-temp-buffer
+    (insert "
+2019-01-01 open Assets:Account1 TDB900
+2019-01-01 open Assets:Account2 TDB900
+2019-01-01 open Assets:Account3 TDB900
+
+2019-01-10  * \"Opening Balances\"
+  Equity:Opening-Balances
+  Assets:Account1        1.00 TDB900
+")
+    (beancount-test-xref-definition-pos "Assets:Account1" 2)
+    (beancount-test-xref-definition-pos "Assets:Account2" 41)
+    (beancount-test-xref-definition-pos "Assets:Account3" 80)))
+
+(ert-deftest beancount/xref-backend-references ()
+  :tags '(xref)
+  (let ((file (make-temp-file "beancount-test-"))
```

Comment:
Why does this test require a buffer baked by a real file, instead than temporary buffer like the other tests?

Reply:
I wanted to do the buffer-backed test but the backend does a call to (buffer-file-name) at some point. While defstruct accepts nil, the older defclasses don't, unfortunately. 

Newer ert (the ert-x.el package) have a macro for this kind of cases but emacs 26.1 doesn't so I came up with this wrapper.

Reply:
It may be time to drop Emacs 26 support. I was looking at integrating the flymake support into `beancount.el` and also that code requires Emacs 27, and I don't want to try to port it to older Emacs.

Reply:
If we keep Emacs 26 support, and we need this work-around, please add a comment explaining why we need it.

Reply:
I'll create a wrapper macro for this

Reply:
Added both a wrapper macro and comments